packages/zowe-explorer/src/trees/dataset/DatasetTree.ts

```diff
@@ -252,6 +252,20 @@ export class DatasetTree extends ZoweTreeProvider<IZoweDatasetTreeNode> implemen
                     Gui.errorMessage(vscode.l10n.t(message));
                     return;
                 }
+
+                // If moving to the same system/lpar but different profile, stop the drag entirely
+                const targetSys = target?.getProfile?.().profile.host;
+                const srcNode = this.draggedNodes[item.uri.path];
+                const sourceSys = srcNode?.getProfile?.().profile.host;
+
+                if (targetSys == sourceSys) {
+                    await Gui.errorMessage(
+                        vscode.l10n.t(
+                            "Cannot move: the source and target are the same dataset on the same system. You're viewing it under a different profile. Any changes you need to view with target profile can be seen via a refresh."
```

Comment:
Suggested message:

Cannot move: The source and target are the same. You are using a different profile to view the target. Refresh to view changes.